docs/proposal-0004-jetstream-preview.md

```diff
@@ -0,0 +1,118 @@
+--end_date                Date for which last analsis should be run
+--generate-population     Generate a population sample (useful if enrollment hasn't happened yet)
+--population-sample-size  The generated population sample size
+--sql-output-dir          Path to write executed SQL 
+--platform                Platform/app to run analysis for
+```
+
+By default the preview would run jetstream on a 3 day analysis window on a population sample of 1%.
+These default parameters can be overwritten when invoking `jetstream preview`.
+
+The resulting data artifacts get written into the `mozdata.tmp` dataset by default, which is configured to delete data after 7 days. Anyone at Mozilla has permissions to write to this dataset.
+
+Once jetstream has generated the preview data it will print out a URL to a Looker dashboard. The Looker dashboard allows users to visualize the preview data. 
+
+The dashboard is a static dashboard that has been added to Looker. It has a filter field that will be pre-filled throught the URL pointing to the temporary datasets that have been generated. It will also have filters so users can select the analysis period, analysis basis, segment, metric and statistic they would like to preview. The visualization will be a line graph with confidence intervals that shows a line for every branch and comparison.
+
+The queries that get executed will be saved locally into files which makes it possible for users to debug the queries (enrollment and metrics) that get executed.
+
+Running the preview command will also print out cost estimates for running analyses on the experiment.
+
+The jetstream CI will need to be updated to publish jetstream to pypi. This way users do not need to pull in the jetstream repository, but can simply run `pip install mozilla-jetstream`.
+
+Jetstream fetches information on how experiments are configured from the Experimenter API. In some cases users might want to get a preview for experiments that haven't been launched yet or that haven't seen any enrollments. For these experiments a random population sample can be generated when specifying `--generate-population` that will be considered as the clients that enrolled.
+
+### Other ideas
+
+Metadata around cost could be stored in a temporary BigQuery table and displayed on the preview dashboard. It might be possible to also store and display the queries that were run as part of the preview including errors that occured. This would help with debugging queries early on.
+
+## Experimenter API
+
+Experimenter only has an API that surfaces experiments that are currently live. To run the preview on experiments that are currently in draft mode, changes would need to be made to surface these experiments through the API.
+
+Having access to experiments in preview would also help users to write and validate configs before the experiment has launched. Currently any pull-request adding a config for an experiment that isn't live cannot be merged or even be validated.
```

Comment:
It doesn't seem like lack of access to draft experiments blocks this work, but would definitely be nice to have. Would exposing draft experiments break anything else? We wouldn't want errors from draft experiments to pop up on the Experimenter UI.

Reply:
We'd fetch draft experiments only when running `jetstream preview`. During normal runs, we won't fetch them, so there shouldn't be any errors from experiments that haven't launched yet